src/main/java/at/medunigraz/imi/bst/trec/query/covid/WordRemovalQueryDecorator.java

```diff
@@ -51,21 +99,155 @@ public WordRemovalQueryDecorator(Query<CovidTopic> decoratedQuery) {
 
     @Override
     public List<Result> query(CovidTopic topic) {
+        String query = topic.getQuery();
+        String question = topic.getQuestion();
         String narrative = topic.getNarrative();
-        Set<String> unfilteredWords = new HashSet<>();
+
+        Set<String> mandatoryWords = new HashSet<>();
+        Set<String> optionalWords = new HashSet<>();
+
+        String[] queryTokens = query.split(" ");
```

Comment:
This line is throwing a `NullPointerException` in `covid/WordRemovalQueryDecoratorTest` because the test doesn't seem to declare a `query`.

Reply:
thx ... just forgot to update the test due to time limitations. it is temporary fixed now, sorry for the trouble